text/metrics/0238-pipeline-monitoring.md

```diff
@@ -0,0 +1,495 @@
+
+In order to use these equations exactly as written, it is important to
+recognize that incoming and outgoing accounting is independent and
+that these signals can lead or lag each order, depending on pipeline
+configuration.
+
+The simplest way to check the invariants, therefore, is to "drain" the
+pipeline so that it is empty, meaning all items have been accounted
+for. For the OpenTelemetry Collector, therefore, it should be
+possible to verify this and warn about improper accounting during
+shutdown. After a pipeline has been drained and the individual
+components shut down, we expect the incoming and outgoing instrument
+values to match exactly.
+
+### Measuring Pipeline Loss Between Segments
+
+Considering an Exporter and Receiver pair connecting two OpenTelemetry
+Collector pipeline segments, we expect ideally the number exported to
+match the number received.
+
+```
+Exported(Preceder) == Received(Follower)
+```
+
+Where Exported and Received counts are defined by the first and second
+equations above, noting that Exported excludes Dropped or Discarded
+outcomes.
+
+When an exporter retries the same item on failures, the equality changes:
+
+```
+Exported(Preceder) <= Received(Follower)
+```
+
+However, some outcomes in the preceder don't indicate that the follow ever 
+received (and therefore couldn't count) the item. This is in the case where a 
+timeout has occurred on the exporter side before any response from the 
+follower. In this case, the equality shifts the other way:
+
+```
+Exported(Preceder) >= Received(Follower)
+```
+
+We can describe exported items that have definitely received a response from 
+the follower by removing `retryable` and `timeout`:
+
+```
+Exported_Responded(Segment) == Exported(Segment) - Retryable(Segment) - Timeout(Segment)
+```
+
+In a real pipeline, therefore, we expect:
+
+```
+Exported_Responded(Preceder) <= Received(Follower)
+```
+
+When an observability platform detects an out-of-range value, there is
+likely to be a defective pipeline or missing instrumentation.
+
+### Measuring Total Pipeline Loss
+
+*WIP: Figure this out. This is a bit subjective. What does an end user expect 
+when calculating total items dropped in failure?*
+
+## Example scenarios
+
+### Agent and Gateway
+
+![image](../img/0238-pipeline-monitoring.png)
+
+![image](../img/0238-pipeline-monitoring-sankey.png)
+
+This example shows multiple pipeline segments with varying outcomes captured in 
+the same diagram. This is focused on one signal, so signal information isn't 
+specified.
+
+*WIP: add details*
+
+### Spanmetrics Connector
+
+![image](../img/0238-pipeline-monitoring-connector.png)
+
+![image](../img/0238-pipeline-monitoring-connector-sankey.png)
+
+This example shows both trace and metric signals sent through an OpenTelemetry 
+Collector with a spanmetrics connector as well as multiple exporters.
+
+*WIP: add details*
+
+### Exporterhelper and batch processor
+
+*WIP: add details*
+
+### Retries
+
+*WIP: add details*
```

Comment:
I think it would be nice if there was some kind of conservation property we could maintain in the presence of retries where we have N errors and need to a single success/failure status.  It seems to me, also, something similar will be applicable for the case of fanout-consumer logic.

Should we have a separate metric for the extra fanout factor which is `N-1` in both of these cases? This number will be needed somewhere to understand pipeline conservation through fanout and retries, I think.